### ISSUE TITLE
Projections no longer working

### ISSUE THREAD

**Issue by Evolva** [EventStore/EventStore#1401]
We run a cluster of 3 windows machines on AWS all of them run EventStore 4.0.2.
Since this morning the projection don't work anymore, they are no longer visible in the UI and call to `/projections/any` seems to timeout.

We got the following error just before the projection stop working
```
PID:01872:021 2017.08.25 14:56:33.050 ERROR QueuedHandlerMRES   ] Error while processing message EventStore.Core.Messages.ClientMessage+ReadStreamEventsForwardCompleted in queued handler 'Projections Master'.
System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at EventStore.Projections.Core.Services.Management.ProjectionManager.OnProjectionsListReadCompleted(ReadStreamEventsForwardCompleted msg, IDictionary`2 registeredProjections, Int64 requestedFrom, Action completedAction) in C:\projects\eventstore\src\EventStore.Projections.Core\Services\Management\ProjectionManager.cs:line 736
   at EventStore.Projections.Core.Services.Management.ProjectionManager.<>c__DisplayClass67_0.<ReadProjectionsList>b__0(ReadStreamEventsForwardCompleted m) in C:\projects\eventstore\src\EventStore.Projections.Core\Services\Management\ProjectionManager.cs:line 710
   at EventStore.Core.Messaging.RequestResponseDispatcher`2.Handle(TResponse message) in C:\projects\eventstore\src\EventStore.Core\Messaging\RequestResponseDispatcher.cs:line 75
   at EventStore.Core.Bus.MessageHandler`1.TryHandle(Message message) in C:\projects\eventstore\src\EventStore.Core\Bus\MessageHandler.cs:line 33
   at EventStore.Core.Bus.InMemoryBus.Publish(Message message) in C:\projects\eventstore\src\EventStore.Core\Bus\InMemoryBus.cs:line 324
   at EventStore.Core.Bus.QueuedHandlerMRES.ReadFromQueue(Object o) in C:\projects\eventstore\src\EventStore.Core\Bus\QueuedHandlerMRES.cs:line 122
```

We try to perform a sequential restart and a full shut down of the cluster with no success.
Right now, a downgrade to 4.0.1 seems to have fix the issue.

Any idea of what cause this issue ?

**Comment by hayley-jean** [comment]
Thank you for reporting this @Evolva 

Would you be willing to send your logs to us? If you don't want to share them here, you can email them to hayley at geteventstore.com.

Also, could you please answer the following:
1. What events are in your `$projections-$all` stream?
2. How many projections do you have running in this cluster?
3. How long was the cluster running on 4.0.2 before this started happening?
4. Did anything change before it happened, such as new projections being added etc?

**Comment by Evolva** [comment]
1. What events are in your $projections-$all stream? 


14 | 14@$projections-$all | $ProjectionDeleted | 2017-08-09 11:59:29 |  
-- | -- | -- | -- | --
13 | 13@$projections-$all | $ProjectionCreated | 2017-08-09 10:53:49 |  
12 | 12@$projections-$all | $ProjectionCreated | 2017-08-09 10:53:14 |  
11 | 11@$projections-$all | $ProjectionDeleted | 2017-08-09 10:52:53 |  
10 | 10@$projections-$all | $ProjectionCreated | 2017-08-09 10:45:50 |  
9 | 9@$projections-$all | $ProjectionCreated | 2017-08-09 10:45:42 |  
8 | 8@$projections-$all | $ProjectionCreated | 2017-01-27 16:16:06 |  
7 | 7@$projections-$all | $ProjectionCreated | 2017-01-06 16:40:48 |  
6 | 6@$projections-$all | $ProjectionCreated | 2016-12-09 10:41:38 |  
5 | 5@$projections-$all | $ProjectionCreated | 2016-12-09 10:41:11 |  
4 | 4@$projections-$all | $ProjectionCreated | 2016-12-09 10:39:00 |  
3 | 3@$projections-$all | $ProjectionCreated | 2016-12-09 10:39:00 |  
2 | 2@$projections-$all | $ProjectionCreated | 2016-12-09 10:39:00 |  
1 | 1@$projections-$all | $ProjectionCreated | 2016-12-09 10:39:00 |  
0 | 0@$projections-$all | $ProjectionsInitialized | 2016-12-09 10:39:00 | JSON

2. How many projections do you have running in this cluster?
- 1 stopped (one time)
- 5 continuous

3. How long was the cluster running on 4.0.2 before this started happening? 
4. Did anything change before it happened, such as new projections being added etc?

4.0.2 was installed less than an hour before the issue
We upgraded to 4.0.2 as we had similar issue on 4.0.0 last week (this time we were abled to disabled/enabled projections and this was enough)

You should have received the log by mail by now.

**Comment by hayley-jean** [comment]
Thank you, I received the mail. We will investigate and let you know what we find

Out of curiosity, do you recall what issue were you running into on 4.0.0?

**Comment by Evolva** [comment]
The issue look similar but we were able to restart the projection with the UI.
Unfortunately, we recycle the instances and we no longer have the logs...

**Comment by hayley-jean** [comment]
Would you be able to send us the result of this curl query?

This will get the data from the events in the `$projections-$all` stream, which will include all the names of your projections. If this counts as sensitive information to your company, you are welcome to email us the result instead of sharing here.

```
curl -i http://localhost:2113/streams/%24projections-%24all\?embed\=tryharder -H "accept:application/json" -u admin:changeit
```

**Comment by Evolva** [comment]
I've send you the result by mail.

On my end I see two strange things:
- The last $ProjectionDeleted seems strange as it has a guid as data instead of a projection name
- Another projection seems to have been created twice

**Comment by hayley-jean** [comment]
Thank you for all your help @Evolva 

The projection that was created twice is the one causing this issue. A PR will be opened to guard against this shortly.

Do you know what version of Event Store was that projection created on?

**Comment by Evolva** [comment]
The projection was created with version 4.0.0  (EventStore-OSS-Win-v4.0.0.zip)

Should we expect another issue in the following days ? If so, is there any workaround ? (dunno delete the stream $projection-$all and recreate all projections)

**Comment by hayley-jean** [comment]
Deleting $projections-$all and recreating projections is unlikely to be a viable solution.
This would cause all of the projections to re-emit their events. If any of your projections had previously emitted to streams, then re-emitting to those same streams would cause them to fault.

Until the fix is released, we would recommend you stay on 4.0.1.
You won't run into any issues because of the duplicate on this version.
